lib/exercism/excerpt.rb

```diff
@@ -0,0 +1,26 @@
+class Excerpt
+  def initialize(text)
+    @html_text = text
+  end
+
+  def limit(length)
+    text[0..(length - 1)] + ellipsis(length)
+  end
+
+  private
+
+  def text
+    @_raw_text ||= begin
+      document = Nokogiri::HTML(@html_text)
+      remove_whitespace(document.text)
+    end
+  end
+
+  def remove_whitespace(string)
+    string.split("\n").map(&:strip).join(' ').strip
+  end
+
+  def ellipsis(length)
+    text.length < length ? '' : '...'
```

Comment:
Yeah, splitting the logic seems kind of awful.

Would you take a look at this and see if it covers this use case? http://makandracards.com/makandra/2579-everything-you-know-about-html_safe-is-wrong I suspect that we can use the SafeBuffer's concatenation operator, but I haven't thought through this and it's been a very long time since I did anything related to this.